### PR TITLE
CPF-296 Treasury Report GraphQL Mutation

### PR DESCRIPTION
**Pull request by Vikariusu**
# Ticket #296 

## Changes
* Added `sendTreasuryReport` GraphQL mutation
* `sendTreasuryReport` triggers aws step function
* TREASURY_STEP_FUNCTION_ARN` env variable is now required to successfully run sendTreasuryReport()

### CODE REVIEW

File: api/src/services/uploads/uploads.ts [hunk 1]
```diff
@@ -258,3 +258,42 @@ export const getUploadsByExpenditureCategory = async () => {
...
+      const input = {
+        reportingPeriod: reportingPeriod.name,
+        organization: organization.name,
+        email: context.currentUser.email,
+        uploadsByExpenditureCategory,
+      }
+
+      await aws.startStepFunctionExecution(
+        arn,
+        undefined,
```
Comment (Vikariusu) [inline api/src/services/uploads/uploads.ts:290]: As per [docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/sfn/command/StartExecutionCommand/): "Optional name of the execution. This name must be unique for your Amazon Web Services account, Region, and state machine for 90 days. For more information, see [Limits Related to State Machine Executions ](https://docs.aws.amazon.com/step-functions/latest/dg/limits.html#service-limits-state-machine-executions)
 in the Step Functions Developer Guide.

If you don't provide a name for the execution, Step Functions automatically generates a universally unique identifier (UUID) as the execution name."

Let me know if we would prefer to name this execution explicitly.

File: api/src/services/uploads/uploads.ts [hunk 2]
```diff
@@ -258,3 +258,42 @@ export const getUploadsByExpenditureCategory = async () => {
 
   return uploadsByExpenditureCategory
 }
+
+export const sendTreasuryReport: MutationResolvers['sendTreasuryReport'] =
```
Comment (as1729) [inline api/src/services/uploads/uploads.ts:262]: @Vikariusu can we add unit tests for this function? I wasn't sure if the comment below that you added is blocking your ability to do so.

Comment (Vikariusu) [reply api/src/services/uploads/uploads.ts:262]: Just added testing

File: api/src/services/uploads/uploads.ts (hunk 1 above)
Comment (as1729) [reply api/src/services/uploads/uploads.ts:290]: I think its fine to leave this alone for now and use the UUID default. 👍

**Review by as1729 — APPROVED** [review]

**Pull request merged**